### PR TITLE
[TTAHUB-279] implement redesigned filter & sorting for the topics bar graph

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Implemented new design for the specialists filter (now a set of checkboxes) and the sorting (the options now match the original design, and the apply filters button is within the dropdown menu).

## How to test

Visit the dashboard. Scroll to the bar graph at the bottom. Observe that the controls for sorting and filtering by specialist are different - the apply filters is confined to each box now. Test them out. Watch the data filter and the graph redraw.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-279

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by jasalisbury — COMMENTED** [review]
Could be my data but I'm seeing some strange behavior
1) Load the dashboard
2) Remove any role from the specialists dropdown, click apply
At this point I can only see four topics. All the topics I no longer see are empty. Also the columns are not as wide after the filter is applied.

Step 1: Screenshot from 2021-08-18 14-50-04

Step 2: Screenshot from 2021-08-18 14-50-16


Also the specialists dropdown does not close on `blur`, the order dropdown does.

**Comment by thewatermethod** [comment]
Thanks @jasalisbury, I believe this is addressed now. 

(To be honest, it was basically working the way I intended it to, but you aren't the first person to suggest that it's a bug and not a feature, so it clearly needed to be changed)

**Comment by jasalisbury** [comment]
Now whenever I unclick a `role` (but before I click apply):
Screenshot from 2021-08-20 09-13-35

**Comment by thewatermethod** [comment]
That's my bad for not testing in Chrome (I guess Firefox's events have different parameters). 

There's an entirely different sort of solution for onBlur now.

**Review by jasalisbury — APPROVED** [review]

File: frontend/src/components/CheckboxSelect.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    ariaLabel,
+  } = props;
+
+  const [checkboxes, setCheckboxes] = useState();
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
+  const [toggleAllChecked, setToggleAllChecked] = useState(toggleAllInitial);
+  const [preventBlur, setPreventBlur] = useState(false);
+
+  useEffect(() => {
+    if (!checkboxes) {
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply frontend/src/components/CheckboxSelect.js]: will do

**Pull request merged**